src/settings.c

```diff
@@ -303,10 +312,95 @@ static void set_key_binding(int *key, const char **bind)
     }
 }
 
-int settings_load(struct user_settings *s, const char *patharg)
+static bool get_settings_path(char *path, unsigned int path_size, const char *patharg)
+{
+    if (patharg != NULL) {
+        snprintf(path, path_size, "%s", patharg);
+        return true;
+    }
+
+    char *user_config_dir = get_user_config_dir();
+    snprintf(path, path_size, "%s%stoxic.conf", user_config_dir, CONFIGDIR);
+    free(user_config_dir);
+
+    /* make sure path exists or is created on first time running */
+    if (!file_exists(path)) {
+        FILE *fp = fopen(path, "w");
```

Comment:
this will fail if the folder does not exist.

Reply:
The config directory is created in main() via `init_default_data_files()`